docs/cadl-for-openapi-dev.md

```diff
@@ -0,0 +1,285 @@
+
+| OpenAPI `type`/`format`           | Cadl type        | Notes                  |
+| --------------------------------- | ---------------- | ---------------------- |
+| `type: integer, format: int32`    | `int32`          |                        |
+| `type: integer, format: int64`    | `int64`          |                        |
+| `type: number, format: float`     | `float32`        |                        |
+| `type: number, format: double`    | `float64`        |                        |
+| `type: string`                    | `string`         |                        |
+| `type: string, format: byte`      |                  |                        |
+| `type: string, format: binary`    | `byte[]`         | any sequence of octets |
+| `type: boolean`                   | `boolean`        |                        |
+| `type: string, format: date`      | `plainDate`      |                        |
+| `type: string, format: date-time` | `zonedDateTime`  | RFC 3339 date          |
+| `type: string, format: password`  | `@secret string` |                        |
+
+OpenAPI supports a variety of "assertions" that can be used further restrict the values allowed for a data type.
+These are actually borrowed into OpenAPI from JSON Schema.
+
+For `type: integer` and `type: number` data types:
+
+| OpenAPI/JSON Schema keyword | Cadl construct              | Notes |
+| --------------------------- | --------------------------- | ----- |
+| `minimum: value`            | `@minimum(value)` decorator |       |
+| `maximum: value`            | `@maximum(value)` decorator |       |
+
+For `type: string` data types:
+
+| OpenAPI/JSON Schema keyword | Cadl construct                | Notes |
+| --------------------------- | ----------------------------- | ----- |
+| `minLength: value`          | `@minLength(value)` decorator |       |
+| `maxLength: value`          | `@maxLength(value)` decorator |       |
+| `pattern: regex`            | `@format(regex)` decorator    |       |
+
+To define a `type: string` parameter or property with an `enum`, define the Cadl element as the union of the enum values.
+For example:
+
+```cadl
+status: "Running" | "Stopped" | "Failed"
+```
+
+## Info Object
+
+In OpenAPI, the `info` object [[v2][v2-info], [v3][v3-info]] contains metadata about the API such as a `title`, `description`, `license`, and `version`.
+
+[v2-info]: https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md#infoObject
+[v3-info]: https://github.com/OAI/OpenAPI-Specification/blob/3.0.3/versions/3.0.3.md#infoObject
+
+In Cadl this information is specified with [decorators on the namespace][cadl-service-metadata].
+
+| OpenAPI `info` field | Cadl decorator    | Notes                    |
+| -------------------- | ----------------- | ------------------------ |
+| `title`              | `@serviceTitle`   |                          |
+| `version`            | `@serviceVersion` |                          |
```

Comment:
Should include the host as part of this metadata: https://github.com/microsoft/cadl/blob/c4c517084f998a3171ed369e42b79941c67f5652/packages/compiler/lib/service.ts#L71-L81